app/models/media_object.rb

```diff
@@ -569,7 +563,48 @@ def move_bulk documents, params
 
   end
 
+  def _update_permalink_and_dependents
```

Comment:
Maybe this should be broken down to update permalink which should be synchronous on before_save and update dependents' permalinks which should be asynchronous on after_save.